sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLListener.scala

```diff
@@ -353,11 +353,19 @@ class SQLListener(conf: SparkConf) extends SparkListener with Logging {
         }
 
         val driverUpdates = executionUIData.driverAccumUpdates.toSeq
-        val totalUpdates = (accumulatorUpdates ++ driverUpdates).filter {
-          case (id, _) => executionUIData.accumulatorMetrics.contains(id)
+        // filter out updates that aren't Longs
```

Comment:
Because all of the SQL UI metrics are longs. We can safely discard anything else without missing SQL metrics. It's as close as we can get to the correct set.